crates/js-component-bindgen/src/ts_bindgen.rs

```diff
@@ -217,36 +222,72 @@ pub fn ts_bindgen(
+                    *
+                    * This function is intended to be similar to the
+                    * `WebAssembly.instantiate` function. The second `imports`
+                    * argument is the \"import object\" for wasm, except here it
+                    * uses component-model-layer types instead of core wasm
+                    * integers/numbers/etc.
+                    *
+                    * The first argument to this function, `compileCore`, is
+                    * used to compile core wasm modules within the component.
+                    * Components are composed of core wasm modules and this callback
+                    * will be invoked per core wasm module. The caller of this
+                    * function is responsible for reading the core wasm module
+                    * identified by `path` and returning its compiled
+                    * `WebAssembly.Module` object. This would use `compileStreaming`
+                    * on the web, for example.
+                    */
+                    export function instantiate(
+                        compileCore: (path: string, imports: Record<string, any>) => Promise<WebAssembly.Module>,
+                        imports: ImportObject,
+                        instantiateCore?: (module: WebAssembly.Module, imports: Record<string, any>) => Promise<WebAssembly.Instance>
+                    ): Promise<{camel}>;
+                ",
+            )
+        }
+
+        Some(InstantiationMode::Sync) => {
+            uwriteln!(
+                bindgen.src,
+                "
+                    /**
+                    * Instantiates this component with the provided imports and
+                    * returns a map of all the exports of the component.
+                    *
+                    * This function is intended to be similar to the
+                    * `WebAssembly.Instantiate` constructor. The second `imports`
+                    * argument is the \"import object\" for wasm, except here it
+                    * uses component-model-layer types instead of core wasm
+                    * integers/numbers/etc.
+                    *
+                    * The first argument to this function, `compileCore`, is
+                    * used to compile core wasm modules within the component.
+                    * Components are composed of core wasm modules and this callback
+                    * will be invoked per core wasm module. The caller of this
+                    * function is responsible for reading the core wasm module
+                    * identified by `path` and returning its compiled
+                    * `WebAssembly.Module` object. This would use the
+                    * `WebAssembly.Module` constructor on the web, for example.
+                    */
+                    export function instantiate(
+                        compileCore: (path: string, imports: Record<string, any>) => WebAssembly.Module,
```

Comment:
In the case of a sync instantiation, we should assume that the modules are already compiled I think, so let's rename this to something like `getCoreModule`?

Reply:
Also now that I'm looking at this, unrelatedly we should probably remove `imports` as a parameter to `compileCore`, as that shouldn't actually be a thing. I can post a follow-up PR if you'd rather do it that way though.

Reply:
Both are done :-).

Reply:
The `imports` argument of the `compileCore` function (now `getCoreModule`) was never used in the tests, only the `imports` argument of `instantiate` was used. It's theoretically a breaking change, but no one should notice it :-/.